.github/workflows/build-windows.yml

```diff
@@ -0,0 +1,297 @@
+  TILEDB_STATIC: OFF
+  TILEDB_TOOLS: ON
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-2016, windows-2019]
+        #considering https://stackoverflow.com/questions/65035256/how-to-access-matrix-variables-in-github-actions
+        environ: [azure, s3, serialization]
+        include:
+          - environ: 'azure'
+            TILEDB_AZURE: ON
+            TILEDB_ARROW_TESTS: ON
+            TILEDB_SERIALIZATION: OFF
+            TILEDB_S3: OFF
+          - environ: 's3'
+            TILEDB_S3: ON
+            TILEDB_ARROW_TESTS: ON
+            TILEDB_SERIALIZATION: OFF
+            TILEDB_AZURE: OFF
+          - environ: 'serialization'
+            TILEDB_S3: OFF
+            TILEDB_AZURE: OFF
+            TILEDB_ARROW_TESTS: OFF
+            TILEDB_SERIALIZATION: ON
+    env:
+      TILEDB_HOME: ${{ github.workspace }}
+      TILEDB_GA_IMAGE_NAME: ${{ matrix.os }}
+      TILEDB_S3: ${{ matrix.TILEDB_S3 }}
+      TILEDB_AZURE: ${{ matrix.TILEDB_AZURE }} #azure }}
+      TILEDB_SERIALIZATION: ${{ matrix.TILEDB_SERIALIZATION }} #serialization }}
+      TILEDB_ARROW_TESTS: ${{ matrix.TILEDB_ARROW_TESTS }}
+    steps:
+      - name: 'tiledb env prep'
+        run: |
+          $env:BUILD_BUILDDIRECTORY = $env:GITHUB_WORKSPACE.replace("TileDB\TileDB","tdbbd") # 't'ile'db' 'b'uild 'd'ir
+          $env:BUILD_SOURCESDIRECTORY = $env:GITHUB_WORKSPACE
+
+          Write-Host "GITHUB_WORKSPACE is '$env:GITHUB_WORKSPACE'"
+          Write-Host "BUILD_BUILDDIRECTORY is '$env:BUILD_BUILDDIRECTORY'"
+          Write-Host "BUILD_SOURCESDIRECTORY is '$env:BUILD_SOURCESDIRECTORY'"
+          Write-Host "GITHUB_ENV is $env:GITHUB_ENV"
+
+          mkdir $env:BUILD_BUILDDIRECTORY
+
+          write-host "BUILD_BUILDDIRECTORY=$env:BUILD_BUILDDIRECTORY" *>> $env:GITHUB_ENV
+          write-host "BUILD_SOURCESDIRECTORY=$env:BUILD_SOURCESDIRECTORY" *>> $env:GITHUB_ENV
+      - name: 'Print env'
+        run: |
+          set
+        shell: cmd
+      - name: ARROW python needs
+        shell: bash
+        if: ${{ matrix.TILEDB_ARROW_TESTS == 'ON' }}
+        run: |
+          set -e pipefail
+          if [[ "$TILEDB_ARROW_TESTS" == "ON" ]]; then
+            pip install pyarrow pybind11 numpy
+          fi
+      - name: Prepare git
+        run: git config --global core.autocrlf false
+      - uses: actions/checkout@v2
+      - name: core tiledb windows build 
+        run: |
+          $ErrorView = "NormalView" 
+          
+          Write-Host "GITHUB_WORKSPACE is '$env:GITHUB_WORKSPACE'"
+          Write-Host "BUILD_BUILDDIRECTORY is '$env:BUILD_BUILDDIRECTORY'"
+          Write-Host "BUILD_SOURCESDIRECTORY is '$env:BUILD_SOURCESDIRECTORY'"
+
+          #directory $env:BUILD_BUILDDIRECTORY was created in step 'tiledb env prep'
+          cd $env:BUILD_BUILDDIRECTORY
+
+          $VSCategory = "Enterprise" # alternate 'Community'
+          if ($env:TILEDB_GA_IMAGE_NAME -eq "windows-2016") {

```

Comment:
One more thing: can go either way I guess, but IMO would avoid hard-coding the bundled paths here, and use either the built-in cmake (3.10 and 3.18 available, I believe), or the `setup-cmake` action: https://github.com/marketplace/actions/actions-setup-cmake

Reply:
I oppose changing use of the hard-coded path(s).

I would not choose to use the pre-installed cmake versions in GA images (3.10, 3.18) because they are not what is generally available with the visual studio (VS) installations, and thus have a mismatch between what is known to be available with VC tools installation and what's running in GA CI (and note that the VS versions do report a custom version id).

I would not choose to use the setup action because, 
A) while it *may* be possible to obtain approx same versions as VS installed versions, because the VS versions report as custom versions (3.12.18081601-MSVC_2, 3.20.21032501-MSVC_2) they would not be exactly the same version as corresponding vanilla cmake versions,
B) even if exactly same version was available it apparently still involves a download and install, which would seem an undesirable time/resource usage.

If it is specifically the hard-coded paths, it would presumably be possible to search programmatically for cmake.exe, but that's still going to require some hard-coded knowledge to be sure the desired cmake.exe is found (that being the one belonging to the expected VS installed version) as well as adding to the time overhead of the run to do this.

Reply:
Ok, marked resolved.